File: simulation_lab/lab_mid.dart
import 'dart:io';

int readInput() {
  int x = int.tryParse(stdin.readLineSync() ?? "") ?? 0;
  return x;
}

void question1() {
  int A = readInput();
  int B = readInput();

  final X = A + B;
  print("X = $X");
}

void question2() {
  int A = readInput();
  int B = readInput();
  int C = readInput();
  int D = readInput();

  final result = A * B - C * D;
  stdout.write("DIFERENCE = $result");
}

void question3() {
  int x = readInput();

  int counter = 0;

  while (x > 0) {
    x = x ~/ 10;
    counter++;
  }

  print("NUmber of digits: $counter");
}

void main(List<String> args) {
  // question1();
  // question2();
  question3();
}

File: simulation_lab/lab_mid.c
#include <stdio.h>
/// question 3
int main()
{
    int a;
    scanf("%d", &a);
    int count = 0;
    while (a > 0)
    {
        a = a / 10;
        count++;
    }
    printf(" NUmber of %d\n", count);
}

#include <stdio.h>
/// question 2
int main()
{
    int A, B, C, D;
    scanf("%d", &A);
    scanf("%d", &B);
    scanf("%d", &C);
    scanf("%d", &D);

    float result = A * B - C * D;

    printf("DIFERENCE = %d\n", result);
}

#include <stdio.h>
/// question 1
int main()
{

    int A, B;
    scanf("%d", &A);
    scanf("%d", &B);
    int X = A + B;
    printf("X = %d\n", X);
}
